source/common/http/utility.cc

```diff
@@ -281,37 +282,65 @@ Utility::parseHttp1Settings(const envoy::api::v2::core::Http1ProtocolOptions& co
   return ret;
 }
 
-void Utility::sendLocalReply(bool is_grpc, StreamDecoderFilterCallbacks& callbacks,
-                             const bool& is_reset, Code response_code, absl::string_view body_text,
-                             const absl::optional<Grpc::Status::GrpcStatus> grpc_status,
-                             bool is_head_request) {
-  sendLocalReply(is_grpc,
+Utility::LocalReplyInfo Utility::generateLocalReplyInfo(const Http::HeaderMap& request_headers) {
+  bool is_grpc = false, is_head_request = false, has_json_content_type = false;
+  // TODO(zyfjeff): This code is copied from Envoy::Grpc::Common::hasGrpcContentType in order to
```

Comment:
What's the circular dependency issue here? There must be some way to share this logic without copying? Please try to have a common function that both places call?

Reply:
Is a dependent circular reference, Http::Utility references grpc::Common(hasGrpcContentType), grpc::Common refers to Http::Utility(getResponseStatus), so a circular reference is generated, so I copied the hasGrpcContentType method directly here. Implementation.


I don't know which file should be placed on the hasGrpcContentType method so that it can be used together by these two places.

Reply:
We should not copy the code, so please create a new file where we can share the logic.